src/server/getters/get-market-price-history.ts

```diff
@@ -0,0 +1,25 @@
+import * as Knex from "knex";
+import { Address, TimestampedPrice, MarketPriceHistory } from "../../types";
+import { sortDirection } from "../../utils/sort-direction";
+
+// Input: MarketID
+// Output: { outcome: [{ price, timestamp }] }
+export function getMarketPriceHistory(db: Knex, marketID: Address|null|undefined, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined, callback: (err?: Error|null, result?: MarketPriceHistory) => void): void {
+  const columnsToSelect: Array<string> = ["outcome", "price", "tradeTime"];
+  let query: Knex.QueryBuilder = db.select(columnsToSelect).from("trades").where({ marketID }).orderBy(sortBy || "tradeTime", sortDirection(isSortDescending, "desc"));
+  if (limit != null) query = query.limit(limit);
+  if (offset != null) query = query.offset(offset);
+  query.asCallback((err?: Error|null, tradesRows?: Array<any>): void => {
+    if (err) return callback(err);
+    if (!tradesRows || !tradesRows.length) return callback(null);
+    const marketPriceHistory: MarketPriceHistory = {};
```

Comment:
I'd think of replacing this whole loop with the more declarative `lodash.groupBy`

https://lodash.com/docs/4.17.4#groupBy